server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v70/RenameOldSonarQubeWayQualityGateTest.java

```diff
@@ -0,0 +1,172 @@
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.internal.TestSystem2;
+import org.sonar.db.CoreDbTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+public class RenameOldSonarQubeWayQualityGateTest {
+
+  private final static long PAST = 10_000_000_000L;
+  private final static long NOW = 50_000_000_000L;
+
+  private final static String SONARQUBE_WAY_QUALITY_GATE = "SonarQube way";
+  private final static String SONAR_WAY_OUTDATED_QUALITY_GATE = "Sonar way (outdated copy)";
+  private final static String SONAR_WAY_QUALITY_GATE = "Sonar way";
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+  @Rule
+  public CoreDbTester db = CoreDbTester.createForSchema(PopulateQualityGatesIsBuiltInTest.class, "quality_gates.sql");
+
+  private System2 system2 = new TestSystem2().setNow(NOW);
+
+  private RenameOldSonarQubeWayQualityGate underTest = new RenameOldSonarQubeWayQualityGate(db.database(), system2);
+
+  @Test
+  public void has_no_effect_if_table_is_empty() throws SQLException {
+    underTest.execute();
+
+    assertThat(db.countRowsOfTable("quality_gates")).isEqualTo(0);
+  }
+
+  @Test
+  public void should_rename_SonarQubeWay_quality_gate() throws SQLException {
+    insertQualityGate(SONARQUBE_WAY_QUALITY_GATE, false);
+
+    underTest.execute();
+
+    assertQualityGates(
+      tuple(SONAR_WAY_OUTDATED_QUALITY_GATE, false, new Date(PAST), new Date(NOW))
+    );
+  }
+
+  @Test
+  public void should_set_builtin_to_false_when_renaming() throws SQLException {
+    insertQualityGate(SONARQUBE_WAY_QUALITY_GATE, true);
+
+    underTest.execute();
+
+    assertQualityGates(
+      tuple(SONAR_WAY_OUTDATED_QUALITY_GATE, false, new Date(PAST), new Date(NOW))
+    );
+  }
+
+  @Test
+  public void should_not_fail_if_a_profile_with_same_name_is_present() throws SQLException {
+    insertQualityGate(SONAR_WAY_OUTDATED_QUALITY_GATE, false);
+    insertQualityGate(SONARQUBE_WAY_QUALITY_GATE, false);
+
+    underTest.execute();
+
+    assertThat(db.countRowsOfTable("quality_gates")).isEqualTo(2);
+  }
+
+  @Test
+  public void should_throw_ISE_if_no_name_is_available() throws SQLException {
+    insertQualityGate(SONARQUBE_WAY_QUALITY_GATE, false);
+    insertQualityGate(SONAR_WAY_OUTDATED_QUALITY_GATE, false);
+    IntStream.range(0, 1000).forEach(
+      i -> insertQualityGate(SONAR_WAY_OUTDATED_QUALITY_GATE + " " + i, false));
+
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("Error during processing of row:");
+    expectedException.expectCause(
+      new CauseMatcher(
+        IllegalStateException.class,
+        "There are too many quality profiles starting with [Sonar way (outdated copy)]"));
+
+    underTest.execute();
+  }
+
+  @Test
+  public void is_reentrant() throws SQLException {
+    insertQualityGate("Whatever", true);
+    insertQualityGate("Whatever2", false);
```

Comment:
test on reentrance should only test... reentrance
Testing that other quality gates are not updated should be done in another test

Reply:
But don't you think that the reentrance should check that other profiles are not touched (before commiting this I had the two methods but found that it was useless) ?

Reply:
In this case only one other profile is enough, IMO
Otherwise the test is testing too many use case
Or add a specific test for that "reentrance_doe_not_updatE_other_qg", but I found this last option a little overkill